esb: Expand ESB_NEVER_DISABLE_TX mode (#22059)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

Enhanced ShockBurst (ESB)
-------------------------

* Updated:

  * By renaming the ``CONFIG_ESB_NEVER_DISABLE_TX`` Kconfig option to :kconfig:option:`CONFIG_ESB_KEEP_TX_IDLE`.

>   * By enabling runtime switching between ACK and non-ACK packets with :kconfig:option:`CONFIG_ESB_KEEP_TX_IDLE` Kconfig option enabled.

Reviewer: ```suggestion
  * By enabling runtime switching between ACK and non-ACK packets using the :kconfig:option:`CONFIG_ESB_KEEP_TX_IDLE` Kconfig option.
```

>   * Renamed :kconfig:option:`CONFIG_ESB_NEVER_DISABLE_TX` Kconfig option to :kconfig:option:`CONFIG_ESB_KEEP_TX_IDLE` Kconfig option.
>   * Extended :kconfig:option:`CONFIG_ESB_KEEP_TX_IDLE` Kconfig option to apply also to packets with acknowledgment (ACK).

Reviewer: ```suggestion
  * By renaming the ``CONFIG_ESB_NEVER_DISABLE_TX`` Kconfig option to :kconfig:option:`CONFIG_ESB_KEEP_TX_IDLE`.
  * By extending the :kconfig:option:`CONFIG_ESB_KEEP_TX_IDLE` Kconfig option to apply also to packets with acknowledgment (ACK).
```

Enhanced ShockBurst (ESB)
-------------------------

* Added:

  * :kconfig:option:`CONFIG_ESB_NEVER_DISABLE_TX` Kconfig option support for packets with ACK.

* Updated:

>   * Renamed :kconfig:option:`CONFIG_ESB_NEVER_DISABLE_TX` to :kconfig:option:`CONFIG_ESB_KEEP_TX_IDLE`.

Reviewer: This does not make much sense. You first add CONFIG_ESB_NEVER_DISABLE_TX and then rename it.

Author: I updated the changelog.
